experimenter/experimenter/nimbus_ui_new/views.py

```diff
@@ -252,11 +270,71 @@ def post(self, *args, **kwargs):
         return response
 
 
+class NimbusExperimentsCloneView(NimbusExperimentViewMixin, RequestFormMixin, CreateView):
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["data"] = kwargs["data"].copy()
+        kwargs["data"]["owner"] = self.request.user
+        kwargs["parent_slug"] = self.kwargs.get("slug")
+        kwargs["branch_slug"] = self.kwargs.get("branch")
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["experiment"] = self.get_object()
+        return context
+
+    def post(self, *args, **kwargs):
+        response = super().post(*args, **kwargs)
+        if response.status_code == 302:
+            response = HttpResponse()
+            response.headers["HX-Redirect"] = reverse(
+                "nimbus-new-detail", kwargs={"slug": self.object.slug}
+            )
+        return response
+
+
+class NimbusExperimentsSidebarCloneView(NimbusExperimentsCloneView):
+    form_class = NimbusExperimentSidebarCloneForm
+    template_name = "nimbus_experiments/clone.html"
+
+
+class NimbusExperimentsPromoteToRolloutView(NimbusExperimentsCloneView):
+    form_class = NimbusExperimentPromoteToRolloutForm
+    template_name = "nimbus_experiments/clone.html"
+
+
+class UpdateCloneSlugView(NimbusExperimentViewMixin, RenderResponseMixin, UpdateView):
+    template_name = "nimbus_experiments/clone_slug_field.html"
+
+    def post(self, request, *args, **kwargs):
+        name = self.request.POST.get("name", "")
+        slug = slugify(name)
+        return self.render_to_response({"slug": slug})
```

Comment:
Oh I didn't realize you were doing this via HTMX too!  Ha!!  Instead of a tiny snippet of JS.  That's so funny.  So in a maximally HTMX world, yes, exactly, all logic lives on the backend, any dynamic updates take place through a full request cycle like this.  Like this is how you'd do it in Phoenix LiveView.  I'd say let's keep it like this and see if this works well for people, or if the latency ends up being too slow in which case we can switch it to some JS, but yeah, this is so fun.  Really good instincts having this take place through a full HTMX cycle.